assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * `modules/analytics-4` data store: audiences.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { isPlainObject, isArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { MODULES_ANALYTICS_4 } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { createValidatedAction } from '../../../googlesitekit/data/utils';
+
+import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';
+const { receiveError, clearError } = errorStoreActions;
+
+const audienceFields = [
+	'displayName',
+	'description',
+	'membershipDurationDays',
+	'eventTrigger',
+	'exclusionDurationMode',
+	'filterClauses',
+];
+
+const audienceRequiredFields = [
+	'displayName',
+	'description',
+	'membershipDurationDays',
+	'filterClauses',
+];
+
+function validateAudience( audience ) {
+	invariant( audience, 'audience is required.' );
+
+	invariant( isPlainObject( audience ), 'Audience must be an object.' );
```

Comment:
The first check is probably redundant because `isPlainObject` will fail if `audience` is falsy.
```suggestion
	invariant( isPlainObject( audience ), 'Audience must be an object.' );
```